src/main.rs

```diff
@@ -224,6 +225,12 @@ fn main() {
         prefix_dir = source_dir;
     }
 
+    let source_root = if source_dir != "" {
```

Comment:
we use the empty PathBuf as a PathBuf in path_rewriting and we just use it as an Option in lib::add_results

Reply:
Yeah, we should switch `rewrite_paths` to use an option too. But in the `rewrite_paths` case the code is clean as it is now as the `source_dir` is always used, instead in `add_results` you're doing `.to_str().unwrap().is_empty()` to check if the `source_dir` is empty, so using an option would be cleaner (and probably faster, since `add_results` is called hundred of thousands of times).